### PR TITLE
peaclock: init at 0.4.3

### PR DESCRIPTION
**Pull request by djanatyn**
###### Motivation for this change
I wanted to use this clock in my terminal. It's cute!

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by djanatyn** [comment]
I tried to get this to build on OS X (since the repository says MacOS is supported) using `gcc8Stdenv`, but it didn't link successfully:

```nix
{ stdenv, lib, fetchFromGitHub, cmake, pkgconfig, libpthreadstubs, icu }:

stdenv.mkDerivation rec {
  pname = "peaclock";
  version = "0.4.3";

  src = fetchFromGitHub {
    owner = "octobanana";
    repo = pname;
    rev = version;
    sha256 = "1582vgslhpgbvcd7ipgf1d1razrvgpq1f93q069yr2bbk6xn8i16";
  };

  nativeBuildInputs = [ cmake ];
  buildInputs = [ libpthreadstubs icu ];

  cmakeFlags = [ "-DCMAKEBUILD_TYPE=release" ] ++ lib.optional (stdenv.isDarwin)
    "-DCMAKE_CXX_COMPILER=${stdenv.cc}/bin/c++";

  enableParallelBuilding = true;

  meta = with stdenv.lib; {
    description = "A clock, timer, and stopwatch for the terminal.";
    homepage = "https://octobanana.com/software/peaclock";
    license = licenses.mit;
    platforms = [ "x86_64-darwin" "x86_64-linux" "i686-linux" ];
    maintainers = with maintainers; [ djanatyn ];
  };
}
```

```diff
+  peaclock = callPackage ../applications/misc/peaclock {
+    stdenv = if stdenv.isDarwin then gcc8Stdenv else stdenv;
+  };
```

Build output:
```
[ 16%] Building CXX object CMakeFiles/peaclock.dir/src/main.cc.o
[ 50%] Building CXX object CMakeFiles/peaclock.dir/src/ob/string.cc.o
[ 50%] Building CXX object CMakeFiles/peaclock.dir/src/peaclock/tui.cc.o
[ 66%] Building CXX object CMakeFiles/peaclock.dir/src/ob/readline.cc.o
[ 83%] Building CXX object CMakeFiles/peaclock.dir/src/peaclock/peaclock.cc.o
[100%] Linking CXX executable peaclock
ld: warning: option -s is obsolete and being ignored
ld: warning: passed two min versions (10.12.0, 10.12) for platform macOS. Using 10.12.
Undefined symbols for architecture x86_64:
  "std::filesystem::__cxx11::path::has_filename() const", referenced from:
      std::filesystem::__cxx11::path::operator/=(std::filesystem::__cxx11::path const&) in main.cc.o
  "std::filesystem::__cxx11::path::lexically_normal() const", referenced from:
      Tui::mkconfig(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) in tui.cc.o
  "std::filesystem::__cxx11::path::has_root_directory() const", referenced from:
      std::filesystem::__cxx11::path::operator/=(std::filesystem::__cxx11::path const&) in main.cc.o
  "std::filesystem::__cxx11::path::compare(std::filesystem::__cxx11::path const&) const", referenced from:
      _main in main.cc.o
      Tui::load_config(std::filesystem::__cxx11::path const&) in tui.cc.o
  "std::filesystem::create_directory(std::filesystem::__cxx11::path const&)", referenced from:
      _main in main.cc.o
  "std::filesystem::status(std::filesystem::__cxx11::path const&)", referenced from:
      _main in main.cc.o
      Tui::mkconfig(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) in tui.cc.o
      Tui::load_config(std::filesystem::__cxx11::path const&) in tui.cc.o
  "std::filesystem::__cxx11::path::_M_split_cmpts()", referenced from:
      std::filesystem::__cxx11::path::path<char [5], std::filesystem::__cxx11::path>(char const (&) [5], std::filesystem::__cxx11::path::format) [clone .isra.274] in main.cc.o
      std::filesystem::__cxx11::path::operator/=(std::filesystem::__cxx11::path const&) in main.cc.o
      std::filesystem::__cxx11::path OB::Parg::get<std::filesystem::__cxx11::path, 0>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in
 main.cc.o
      _main in main.cc.o
      Tui::mkconfig(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) in tui.cc.o
      Tui::load_config(std::filesystem::__cxx11::path const&) in tui.cc.o
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/peaclock.dir/build.make:163: peaclock] Error 1
make[1]: *** [CMakeFiles/Makefile2:95: CMakeFiles/peaclock.dir/all] Error 2
make: *** [Makefile:149: all] Error 2
builder for '/nix/store/g6wdj74dx7s5wjddz69v7z8782wdnjhy-peaclock-0.4.3.drv' failed with exit code 2
error: build of '/nix/store/g6wdj74dx7s5wjddz69v7z8782wdnjhy-peaclock-0.4.3.drv' failed
```

**Comment by djanatyn** [comment]
I was able to build successfully on pinephone, so I added `aarch64-linux` support.

File: pkgs/top-level/all-packages.nix
Comment (r-burns) [inline pkgs/top-level/all-packages.nix:22733]:
```suggestion
  peaclock = callPackage ../applications/misc/peaclock {
    stdenv = gccStdenv;
  };
```

File: pkgs/applications/misc/peaclock/default.nix
Comment (r-burns) [inline pkgs/applications/misc/peaclock/default.nix:23]:
```suggestion
    platforms = platforms.unix;
```

File: pkgs/applications/misc/peaclock/default.nix
Comment (r-burns) [inline pkgs/applications/misc/peaclock/default.nix:1]:
```suggestion
{ stdenv, fetchFromGitHub, cmake, libpthreadstubs, icu }:
```
looks to be unused

**Review by r-burns — CHANGES_REQUESTED** [review]
hehe it is cute!

I was able to run on macos with these changes, and the LC_ALL=C workaround from https://github.com/octobanana/peaclock/issues/16. It looks like gccStdenv is gcc 9, which has a fully functioning stdc++fs. Although clang 7 is supposed to work, so maybe that would be preferable?

**Comment by djanatyn** [comment]
I tried `stdenvClang`:

```
[ 16%] Building CXX object CMakeFiles/peaclock.dir/src/ob/string.cc.o
[ 33%] Building CXX object CMakeFiles/peaclock.dir/src/peaclock/tui.cc.o
[ 50%] Building CXX object CMakeFiles/peaclock.dir/src/main.cc.o
[ 66%] Building CXX object CMakeFiles/peaclock.dir/src/ob/readline.cc.o
[ 83%] Building CXX object CMakeFiles/peaclock.dir/src/peaclock/peaclock.cc.o
[100%] Linking CXX executable peaclock
ld: warning: option -s is obsolete and being ignored
ld: warning: passed two min versions (10.12.0, 10.12) for platform macOS. Using 10.12.
ld: warning: ignoring file CMakeFiles/peaclock.dir/src/main.cc.o, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file format ( 0xDE 0x
C0 0x17 0x0B 0x00 0x00 0x00 0x00 0x14 0x00 0x00 0x00 0xF4 0x65 0x08 0x00 )
ld: warning: ignoring file CMakeFiles/peaclock.dir/src/ob/string.cc.o, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file format ( 0x
DE 0xC0 0x17 0x0B 0x00 0x00 0x00 0x00 0x14 0x00 0x00 0x00 0xD8 0xB6 0x01 0x00 )
ld: warning: ignoring file CMakeFiles/peaclock.dir/src/peaclock/tui.cc.o, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file format (
 0xDE 0xC0 0x17 0x0B 0x00 0x00 0x00 0x00 0x14 0x00 0x00 0x00 0x54 0x35 0x0E 0x00 )
ld: warning: ignoring file CMakeFiles/peaclock.dir/src/ob/readline.cc.o, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file format (
0xDE 0xC0 0x17 0x0B 0x00 0x00 0x00 0x00 0x14 0x00 0x00 0x00 0x80 0xB5 0x05 0x00 )
ld: warning: ignoring file CMakeFiles/peaclock.dir/src/peaclock/peaclock.cc.o, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file for
mat ( 0xDE 0xC0 0x17 0x0B 0x00 0x00 0x00 0x00 0x14 0x00 0x00 0x00 0x48 0x44 0x06 0x00 )
Undefined symbols for architecture x86_64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [CMakeFiles/peaclock.dir/build.make:163: peaclock] Error 1
make[1]: *** [CMakeFiles/Makefile2:95: CMakeFiles/peaclock.dir/all] Error 2
make: *** [Makefile:149: all] Error 2
builder for '/nix/store/hkmlb9b88vm01mlvh86dl5f18g0x4zci-peaclock-0.4.3.drv' failed with exit code 2
error: build of '/nix/store/hkmlb9b88vm01mlvh86dl5f18g0x4zci-peaclock-0.4.3.drv' failed
```

No worries - your changes worked and I got peaclock running on my macbook 🎉 Thank you @r-burns!

I was looking at `makeWrapper` for setting `LC_ALL=C` on certain platforms. However, I think it might be best to leave that to be fixed in upstream given that it affects all musl distributions?

If so, I think this is working on all platforms based on my testing.

File: pkgs/applications/misc/peaclock/default.nix
Comment (IvarWithoutBones) [inline pkgs/applications/misc/peaclock/default.nix:17]: This is not required with CMake, as it is enabled by default. See [this section of the manual](https://nixos.org/manual/nixpkgs/stable/#ssec-stdenv-attributes) for more information.

File: pkgs/applications/misc/peaclock/default.nix
Comment (IvarWithoutBones) [inline pkgs/applications/misc/peaclock/default.nix:20]: `meta.description` should not end with a period, according to the [contributing guidelines](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes).

**Comment by AndersonTorres** [comment]
Now squash the 5 commits to 2 only (the one for the new package maintainer and the other for the package itself).

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
Added a few suggestions :)

**Comment by AndersonTorres** [comment]
Also, squash the commits.

File: pkgs/applications/misc/peaclock/default.nix
Comment (djanatyn) [reply pkgs/applications/misc/peaclock/default.nix]: Removed the period.

**Comment by djanatyn** [comment]
Commits squashed, review comments resolved. Thanks for walking me through this, it's so exciting contributing to something I use everyday.

**Comment by AndersonTorres** [comment]
LGTM, LGTBorg.

**Pull request merged**
